src/core/manifest.js

```diff
@@ -69,34 +68,46 @@ async function buildManifest (env, opts = {}) {
     }
 
     if (stats.isFile()) {
+      // Only watch for certain file extensions (defaults to *.md)
       if (opts.extensions && opts.extensions.indexOf(ext) === -1) {
         return null
       }
 
+      // Warn if there are any route index conflicts
       await warnForIndexConflict(path)
       await checkForIndexConflict(path)
 
+      // @TODO: Check fro Readme.md or [folder].md
       const isIndex = /\/index\.[\w]+$/.test(path)
       const shouldGetContent = env === 'production'
-      const hydrated = await hydrate(path, dir, opts.outputDir, shouldGetContent)
+      const frontMatter = await getFrontmatter(path)
+      const hydrated = await hydrate(
+        frontMatter, path, dir, opts.outputDir, shouldGetContent, config
+      )
 
+      // Ignore files with a `draft` status
+      if (frontMatter.draft) return
+
+      // Detect duplicate URLs
       if (urlmap[hydrated.url]) {
         const duplicated = files[urlmap[hydrated.url]].file
-        throw new Error(`Can't use a URL more than once: ${hydrated.url}\n\t- ${duplicated}\n\t- ${hydrated.file}`)
+        throw new Error(
+          `Can't use a URL more than once: ${hydrated.url}\n\t- ${duplicated}\n\t- ${hydrated.file}`
+        )
       }
 
       filemap[path] = files.push(hydrated) - 1
       urlmap[hydrated.url] = filemap[path]
 
-      if (hydrated.draft) {
-        return
-      }
-
       return isIndex ? {
         indexLink: hydrated.url,
+        order: hydrated.order,
```

Comment:
Why does this need to be in the actual manifest? It should already be sorted by the time it gets passed to the theme app.